src/app/modules/item/components/chapter-children/chapter-children.component.ts

```diff
@@ -30,7 +56,23 @@ export class ChapterChildrenComponent implements OnChanges, OnDestroy {
         .get(this.itemData.item.id, this.itemData.currentResult.attemptId)
         .subscribe(
           children => {
-            this.children = children;
+            this.children = children.map(child => {
+              const res = bestAttemptFromResults(child.results);
+              return {...child,
+                isLocked: !canCurrentUserViewItemContent(child),
+                result: res === null ? undefined : {
+                  is_validated: res.validated,
+                  current_score: res.score,
+                },
+              };
+            });
+
+            this.case = 'normal';
+            if (this.children.length === 0)
+              this.case = 'empty';
+            else if (this.children.filter(item => item.result && item.category === 'Validation' && !item.result.is_validated).length === 0)
```

Comment:
But that's not really correct as you count those without result as "validated".

Reply:
Actually, if all validation children are validated, this is correct to claim that the chapter is validated, but that's what the backend computes itself. So using the "validated" from the chapter directly is better. 
But now... what if the chapter is not validated while all children are validated (bug or because there are children which are not currently visible by the user)? For now, you cannot really do anything except maybe not displaying the sentence...

Reply:
Ok, so I have proposed (in another remark) the new states and so you should be:
-  'ready' (so displaying msg) is this chapter is validated OR every children with category === 'Validation' has a result and is validated
- 'ready-missing-validation'` is other cases.

Reply:
fixed